src/functions/module_statistical.ts

```diff
@@ -555,6 +772,99 @@ export const COVARIANCE_S = {
   isExported: true,
 } satisfies AddFunctionDescription;
 
+// -----------------------------------------------------------------------------
+// FORECAST
+// -----------------------------------------------------------------------------
+export const FORECAST: AddFunctionDescription = {
+  description:
+    "Calculates the expected y-value for a specified x based on a linear regression of a dataset.",
+  args: [
+    arg("x (number)", _t("The value on the x-axis to forecast.")),
+    arg(
+      "data_y (range<number>)",
+      _t("The range representing the array or matrix of dependent data.")
+    ),
+    arg(
+      "data_x (range<number>)",
+      _t("The range representing the array or matrix of independent data.")
+    ),
+  ],
+  returns: ["NUMBER"],
+  compute: function (
+    x: Maybe<CellValue>,
+    dataY: Matrix<CellValue>,
+    dataX: Matrix<CellValue>
+  ): number {
+    const _dataY = matrixMap(dataY, (value) => toNumber(value, this.locale));
+    const _dataX = matrixMap(dataX, (value) => toNumber(value, this.locale));
```

Comment:
I think we don't want to cast possible value into a number for matrix, 
We must ingnot all non-number type

e.g. with covarience
https://github.com/odoo/o-spreadsheet/pull/2557/files#diff-adaa069c2f4558d87ba4a7eb21d32da0f1f37b8925e0a88cb64dcf59b99396baR48

Reply:
apply this comment to all formulas